website/docs/docs/build/unit-tests.md

```diff
@@ -0,0 +1,256 @@
+
+),
+	
+check_valid_emails as (
+
+    select
+        customers.customer_id,
+        customers.first_name,
+        customers.last_name,
+	coalesce (regexp_like(
+            customers.email, '^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}$'
+        )
+        = true
+        and accepted_email_domains.tld is not null,
+        false) as is_valid_email_address
+    from customers
+		left join accepted_email_domains
+        on customers.email_top_level_domain = lower(accepted_email_domains.tld)
+
+)
+
+select * from check_valid_emails
+```
+</file>
+
+The logic posed in this example can be challenging to validate. You can add a unit test to this model to ensure the `is_valid_email_address` logic captures all known edge cases: emails without `.`, emails without `@`, and emails from invalid domains.
+
+<file name='dbt_project.yml'> 
+
+```yaml
+unit_tests:
+  - name: test_is_valid_email_address
+    description: "Check my is_valid_email_address logic captures all known edge cases - emails without ., emails without @, and emails from invalid domains."
+    model: dim_customers
+    given:
+      - input: ref('stg_customers')
+        rows:
+          - {customer_id: 1, email: cool@example.com, email_top_level_domain: example.com}
+          - {customer_id: 2, email: cool@unknown.com, email_top_level_domain: unknown.com}
+          - {customer_id: 3, email: badgmail.com, email_top_level_domain: gmail.com}
+          - {customer_id: 4, email: missingdot@gmailcom, email_top_level_domain: gmail.com}
+      - input: ref('top_level_email_domains')
+        rows:
+          - {tld: example.com}
+          - {tld: gmail.com}
+    expect:
+      rows:
+        - {customer_id: 1, is_valid_email_address: true}
+        - {customer_id: 2, is_valid_email_address: false}
+        - {customer_id: 3, is_valid_email_address: false}
+        - {customer_id: 4, is_valid_email_address: false}
+
+```
+</file>
+
+The previous example defines the mock data using the inline `dict` format, but you can also use `csv` either inline or in a separate fixture file. 
+
+You only have to define the mock data for the columns you care about. This enables you to write succinct and _specific_ unit tests.
+
+:::note
+
+The direct parents of the model that you’re unit testing (in this example, `stg_customers` and `top_level_email_domains`) need to exist in the warehouse before you can execute the unit test.
+
+Use the `--empty` flag to build an empty version of the models to save warehouse spend. 
+
+```bash
+
+dbt run —-select "stg_customers top_level_email_domains" --empty
+
+```
+
+Alternatively, use `dbt build` to, in lineage order:
+
+- Run the unit tests on your model.
+- Materialize your model in the warehouse.
+- Run the data tests on your model.
+
+:::
+
+Now you’re ready to run this unit test. You have a couple of options for commands depending on how specific you want to be: 
+
+- `dbt test —-select dim_customers` runs _all_ of the tests on `dim_customers`.
```

Comment:
```suggestion
- `dbt test --select dim_customers` runs _all_ of the tests on `dim_customers`.
```

Formatting is weird for all of the `--` on this page - we should go through and update them

![Screenshot 2024-02-14 at 11 06 42 AM](https://github.com/dbt-labs/docs.getdbt.com/assets/53586774/dced3b18-0b81-46ed-97aa-6b11f48b5a6c)